src/FakeItEasy/CallbackConfigurationExtensions.Generated.tt

```diff
@@ -0,0 +1,70 @@
+<#@ template hostspecific="false" language="C#" #>
+<#@ output extension=".cs" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+namespace FakeItEasy
+{
+    using System;
+#if FEATURE_NETCORE_REFLECTION
+    using System.Reflection;
+#endif
+
+    using FakeItEasy.Configuration;
+
+    public static partial class CallbackConfigurationExtensions
+    {
+<#
+    for (int typeParametersCount = 1; typeParametersCount <= 8; typeParametersCount++)
+    {
+        var typeParamList = string.Join(", ", Enumerable.Range(1, typeParametersCount).Select(i => "T" + i));
+        var argumentList = string.Join(", ", Enumerable.Range(1, typeParametersCount).Select(i => $"call.GetArgument<T{i}>({i - 1})"));
+#>
+        /// <summary>
+        /// Executes the specified action when a matching call is being made.
+        /// </summary>
+        /// <param name="configuration">The configuration that is extended.</param>
+        /// <param name="actionToInvoke">The <see cref="Action{T1}"/> to invoke.</param>
+        /// <typeparam name="TInterface">The type of configuration interface to return.</typeparam>
+<#
+        for (int i = 1; i <= typeParametersCount; i++)
+        {
+#>
+        /// <typeparam name="T<#= i #>">The type of the <#= Ordinals[i] #> argument of the faked method call.</typeparam>
+<#
+        }
+#>
+        /// <exception cref="FakeConfigurationException">The signatures of the faked method and the <paramref name="actionToInvoke"/> do not match.</exception>
+        /// <returns>The fake object.</returns>
+        public static TInterface Invokes<TInterface, <#= typeParamList #>>(this ICallbackConfiguration<TInterface> configuration, Action<<#= typeParamList #>> actionToInvoke)
+        {
+            Guard.AgainstNull(configuration, nameof(configuration));
+            Guard.AgainstNull(actionToInvoke, nameof(actionToInvoke));
+
+            return configuration.Invokes(call =>
+                {
+                    ValueProducerSignatureHelper.AssertThatValueProducerSignatureSatisfiesCallSignature(call.Method, actionToInvoke.GetMethodInfo(), NameOfInvokesFeature);
+
+                    actionToInvoke(<#= argumentList #>);
+                });
+        }
+
+<#
+    }
+#>
+    }
+}
+<#+
+private static readonly string[] Ordinals =
+{
+    "", // zero

```

Comment:
could be `"zeroth"`. Avoids the comment and could be helpful if it accidentally leaks out.
Overall, I really like the use of the `Ordinals` array.

Reply:
I thought about it, but thought it wouldn't really make sense as we're not supposed to use it... I can change it anyway if you prefer.

Reply:
I think I'd feel better. If it won't kill you, maybe do?
We prefer `string.Empty` to `""` anyhow, so it's not like you won't be touching this line regardless. :wink: